### PR TITLE
2017-11-21作业

### PR DESCRIPTION
**Pull request by Ccrool**

### CODE REVIEW

File: hja/2017-11-17/.idea/2017-11-17.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (calidion) [inline hja/2017-11-17/.idea/2017-11-17.iml]: 避免提交.idea这样的文件或者文件夹

File: hja/2017-11-21/demo-master/data/mail.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
+{
```
Comment (calidion) [inline hja/2017-11-21/demo-master/data/mail.json]: 数据文件或者目录不必提交

**Comment by Ccrool** [comment]
删除了data和idea

File: hja/2017-11-21/demo-master/entities/user.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+function User(username, password) {
+    this.username = username;
+    this.email = username;
+    this.password = password;
+}
+
+User.register = function (socket, username, password,cb) {
+    console.log('现在在注册函数\n');
+    storage.read(function(error,UsersInfo){
```
Comment (calidion) [inline hja/2017-11-21/demo-master/entities/user.js]: storage.read(function(error,UsersInfo){
=>
storage.read(function(error, UsersInfo) {

File: hja/2017-11-21/demo-master/entities/user.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+       if(error){
+           cb(error);
+           return;
+       }
+       if(UsersInfo[username]){
```
Comment (calidion) [inline hja/2017-11-21/demo-master/entities/user.js]: if(UsersInfo[username]){
=>
if(UsersInfo[username]) {

File: hja/2017-11-21/demo-master/entities/user.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+           console.log('已经存在');
+           cb(true);
+           return;
+       }
+       UsersInfo[username] = {
+           // socket:socket,
+           user:new User(username,password)
```
Comment (calidion) [inline hja/2017-11-21/demo-master/entities/user.js]: user:new User(username,password)
=>
user: new User(username,password)

File: hja/2017-11-21/demo-master/entities/user.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    console.log("user manager login");
+
+    storage.read(function (error,UsersInfo) {
+        if(error){
+            cb(error);
+            return;
+        }
+        if(UsersInfo[username]){
+            console.log('存在');
+            if(UsersInfo[username].user.username===username && UsersInfo[username].user.password === password){
```
Comment (calidion) [inline hja/2017-11-21/demo-master/entities/user.js]: UsersInfo[username].user.username===username
=>
UsersInfo[username].user.username === username

File: hja/2017-11-21/demo-master/entities/user.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+                //---此时往sockets里加东西
+                sockets.push({
+                    socket:socket,
+                    nowUser:username
+                });
+                console.log(sockets);
+                return;
+            }
+            else{
+                cb(true);
```
Comment (calidion) [inline hja/2017-11-21/demo-master/entities/user.js]: }
            else{
=>
} else {

**Comment by Ccrool** [comment]
格式修改

File: hja/2017-11-17/今天课上的/machine/Man/Mail.js [hunk 8]
```diff
@@ -1,5 +1,29 @@
 const User = require('./user').User;
-let mails = {};
+const fs = require('fs');
+const filename = '../data/mail.json';
+
+// let mails = {};
+Mail.readJSON = function (filename,callback) {
+        let data = [];
+        const rs = fs.createReadStream(filename);
+        rs.on('data',function (chunk) {
```
Comment (calidion) [inline hja/2017-11-17/今天课上的/machine/Man/Mail.js]: rs.on('data',function (chunk) {
=>
rs.on('data', function (chunk) {

File: hja/2017-11-17/今天课上的/machine/Man/Mail.js [hunk 9]
```diff
@@ -1,5 +1,29 @@
...
+            data.push(chunk);
+        });
+        rs.on('end',function () {
+            let json = String(Buffer.concat(data,data.length));
```
Comment (calidion) [inline hja/2017-11-17/今天课上的/machine/Man/Mail.js]: let json = String(Buffer.concat(data,data.length));
=>
 let json = String(Buffer.concat(data, data.length));

File: hja/2017-11-17/今天课上的/machine/Man/Mail.js [hunk 10]
```diff
@@ -1,5 +1,29 @@
...
+            callback(json);
+        });
+};
+
+Mail.writeJSON = function (filename,json,callback) {
```
Comment (calidion) [inline hja/2017-11-17/今天课上的/machine/Man/Mail.js]: 同上

File: hja/2017-11-17/今天课上的/machine/Man/Mail.js [hunk 11]
```diff
@@ -9,6 +33,7 @@ function Mail(sender, receiver, title, body) {
 }
 
 Mail.send = function (sender, receiver, title, body) {
+    let mails = this.readJSON(filename,);
```
Comment (calidion) [inline hja/2017-11-17/今天课上的/machine/Man/Mail.js]: let mails = this.readJSON(filename,);
=》
let mails = this.readJSON(filename);

**Comment by Ccrool** [comment]
重新提交17

**Pull request merged**